meshnets/utils/datasets.py

```diff
@@ -1,86 +1,113 @@
-
-    Note that the files do not have to be named like this.
-
-    """
-
-    def __init__(self, data_dir):
-        super().__init__()
-        self.root_dir = data_dir
-        self.files = sorted(os.listdir(data_dir))
-
-    def __len__(self):
-        return len(self.files)
-
-    def __getitem__(self, idx):
-        """The get is simply a method to retrieve an element from the
-        dataset. The only thing that exists in RAM at this point are
-        the paths to the examples. Later, this Dataset class is
-        wrapped by pytorch DataLoaders. These dataloaders will be
-        responsible for batching the data and feeding the batches to
-        the model.
-
-        Example:
-            >>> dataset = FromDiskDataset('data_dir')
-            >>> dataloader = DataLoader(dataset, batch_size=32)
-            >>> for batch in dataloader:
-            >>>     x, y = batch, batch.y
-        """
-        data_path = os.path.join(self.root_dir, self.files[idx])
-        return torch.load(data_path)
+# TODO(victor): the wind vector should be
+# obtained individually for each mesh
+WIND_VECTOR = (10, 0, 0)
 
 
 class FromDiskGeometricDataset(Dataset):
     """Reads a torch_geometric dataset from a given directory.
     
+    The data is loaded from disk each time it as accessed to reduce
+    RAM requirements. The data directory is assumed to be structured
+    as follow:
+
+    data/
+        sample_1/
+            mesh.vtk
+            graph.pt
+        sample_2/
+            mesh.vtk
+            graph.pt
+        sample_3/
+            mesh.vtk
+            graph.pt
+
+    Note that the directory, file names, and file extensions can be different.
+    `process_data` allows to produce the graph files upon instanciating the 
+    dataset class from a folder containing only the mesh files.
+
     This class inherits from torch_geometric Dataset and implements its
-    abstract methods `len` and `get` from the FromDiskDataset class.
-    This offers acces to the torch_geometric Dataset attributes when the .pt
-    files retrieved from disk are torch_geometric Data.
+    abstract methods `len` and `get`. This offers acces to the torch_geometric
+    Dataset attributes when the .pt files retrieved from disk are
+    torch_geometric Data.
 
     It also offer a new property called 'num_label_features' following
     the implementation of 'num_node_features' and 'num_edge_features'
     in torch_geometric Dataset.
     """
 
-    def __init__(self, data_dir, *args, **kwargs):
+    def __init__(self,
+                 data_dir: str,
+                 *args,
+                 process_data: bool = False,
+                 mesh_file_ext: str = '.vtk',
+                 graph_file_ext: str = '.pt',
+                 **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self.from_disk_dataset = FromDiskDataset(data_dir=data_dir)
+
+        self.root_dir = data_dir
+        self.samples = sorted(os.listdir(data_dir))
```

Comment:
I'd count only subdirs of `data_dir` as samples. You may want to have a dataset summary file, e.g. `data_summary.txt` in `data_dir`.